src/components/tx-flow/flows/NewSpendingLimit/CreateSpendingLimit.tsx

```diff
@@ -46,15 +36,22 @@ export const _validateSpendingLimit = (val: string, decimals?: number) => {
   }
 }
 
-export const SpendingLimitForm = ({ data, onSubmit }: Props) => {
+export const CreateSpendingLimit = ({
+  params,
+  onSubmit,
+}: {
+  params: NewSpendingLimitFlowProps
+  onSubmit: (data: NewSpendingLimitFlowProps) => void
+}) => {
   const chainId = useChainId()
   const [showResetTime, setShowResetTime] = useState<boolean>(false)
   const { balances } = useVisibleBalances()
 
   const resetTimeOptions = useMemo(() => getResetTimeOptions(chainId), [chainId])
+  const defaultResetTime = resetTimeOptions[0].value
```

Comment:
An idea for when we move on to refactoring this component. If the user selects a reset time and goes back to this step it should be active and show the previously selected option instead of being closed and resetting to the default value.

Reply:
Good point. I've adjusted this to preserve the value.